src/main/java/duke/task/TaskStorage.java

```diff
@@ -0,0 +1,178 @@
+package duke.task;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.Duke;
+
+/**
+ * Duke.TaskStorage class is used to store all the tasks in the list. It also has the capability of saving the tasks to
+ * a file and loading the tasks from a file.
+ *
+ * The task list class and the storage class are not separated because I disagree with the idea of having a separate
+ * storage class in this project since the storage class is not reusable and is tightly coupled with the task list.
+ */
+public class TaskStorage {
+    private final int size = 100;
+    private final String filePath = "data/meowies.txt";
+    private final ArrayList<Task> tasks = new ArrayList<>(size);
+    private final File file;
+
+    /**
+     * Loads the tasks from the file.
+     */
+    public TaskStorage() {
+        this.file = new File(filePath);
+        try {
+            this.loadFromFile();
+        } catch (FileNotFoundException | Duke.WrongFormatException | Duke.InvalidFileException e) {
+            try {
+                this.file.getParentFile().mkdirs();
+                this.file.createNewFile();
+            } catch (Exception ex) {
+                System.out.println(ex.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Clean constructor for testing.
+     * @param isTest Whether the program is running in test mode.
+     */
+    public TaskStorage(boolean isTest) {
+        String newFilePath = "data/meowies_test.txt";
+        this.file = new File(newFilePath);
+        try {
+            this.loadFromFile();
+        } catch (FileNotFoundException | Duke.WrongFormatException | Duke.InvalidFileException e) {
+            try {
+                this.file.getParentFile().mkdirs();
+                this.file.createNewFile();
+            } catch (Exception ex) {
+                System.out.println(ex.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Loads the tasks from the file.
+     * @param input The input string.
+     * @return The output string.
+     */
+    public String save(String input) {
+        Task task;
+        try {
+            task = Task.createTask(input);
+        } catch (Duke.WrongCommandException | Duke.WrongFormatException e) {
+            return e.getMessage();
+        }
+
+        this.tasks.add(task);
+        this.saveToFile();
+
+        return "added: " + task;
+    }
+
+    /**
+     * Mark a task as done.
```

Comment:
The first sentence of the JavaDoc comment should start with Returns, Sends, Adds instead of Return, Send, Add. Perhaps change this to "Marks a task as done".